src/components/ActionBar.tsx

```diff
@@ -1,74 +1,56 @@
     updateInfo: boolean;
+    setModal: Dispatch<SetStateAction<Modals>>;
 };
 
 function ActionBar(props: ActionBarProps) {
-    const { smartWallets, setSmartWallets, updateInfo } = props;
+    const { setSmartWallets, updateInfo, setModal } = props;
 
     const [workerBalance, setWorkerBalance] = useState('0');
 
-    const { state, dispatch } = useStore();
+    const { state } = useStore();
 
-    const setBalance = async (
-        smartWallet: SmartWallet
-    ): Promise<SmartWalletWithBalance> => {
-        const balance = await Utils.tokenBalance(
-            smartWallet.address,
-            state.token!.address
-        );
-        const rbtcBalance = await Utils.getBalance(smartWallet.address);
-        const swWithBalance = {
-            ...smartWallet,
-            balance: `${Utils.fromWei(balance)} ${state.token!.symbol}`,
-            rbtcBalance: `${Utils.fromWei(rbtcBalance)} RBTC`
-        };
-        return swWithBalance;
-    };
-
-    useEffect(() => {
-        if (!state.account || !state.provider || !state.token) {
-            return;
-        }
-        (async () => {
-            let index: number = 0;
-            let found: boolean = true;
-            const tempSmartWallets: SmartWalletWithBalance[] = [];
-            while (found) {
-                // eslint-disable-next-line no-await-in-loop
-                const swAddress = await state.provider!.generateSmartWallet(
-                    index + 1
+    const loadSmartWallets = async () => {
+        let tempSmartWallets: SmartWallet[] = [];
+        try {
+            if (state.account in localStorage) {
+                tempSmartWallets = JSON.parse(
+                    localStorage.getItem(state.account)!
                 );
-                // eslint-disable-next-line no-await-in-loop
-                const deployed = await state.provider!.isSmartWalletDeployed(
-                    swAddress.address
-                );
-                if (deployed) {
-                    // eslint-disable-next-line no-await-in-loop
-                    const smartWalletWithBalance = await setBalance(swAddress);
-                    tempSmartWallets.push(smartWalletWithBalance);
-                    index += 1;
-                } else {
-                    setSmartWallets(tempSmartWallets);
-                    found = false;
-                }
             }
-        })();
-    }, [state.account, state.token, updateInfo]);
+        } catch (e) {
+            console.log(
+                'Failed trying to read smart wallets, erased all previous smart wallets'
+            );
+            console.log(e);
+        }
+        console.log(tempSmartWallets);
+        const tempSmartWalletsWithBalance: SmartWalletWithBalance[] = [];
+        for (let i = 0; i < tempSmartWallets.length; i += 1) {
+            // eslint-disable-next-line no-await-in-loop
+            const tempSmartWallet = await Utils.setBalance(
```

Comment:
I know this has been introduced in a different PR, but I'll fix it now, but calling this method in parallel.